docs/testing/e2e-plan.md

```diff
@@ -0,0 +1,79 @@
+
+### Long Term
+
+Following the short term goals, several broader, more in-depth changes will help our end-to-end tests further raise our confidence. What follows is a brief summary of the larger components or actions required, with a more detailed design after the short term priorities have been satisfied.
+
+#### Test Manifests
+
+A list of each artifact produced, along with identifiers and URLs to acquire each artifact for testing. This manifest will serve as one input to a new testing framework that will be able to ensure that the desired combinations of providers, host OSes, and any other test dimensions deemed appropriate are run.
+
+This manifest will be generated in response to a pull request change or commit to selected branches (main, release-x, etc) of the eks-anywhere-packages, eks-anywhere, and eks-anywhere-build-tooling repositories (the last of which possibly limited to changes affecting a curated packages project), and will in turn be used by the testing framework.
+
+#### Increased Frequency Of E2E Test Runs
+
+If feasible, end-to-end tests will be run on each (ok-to-test) pull request on the eks-anywhere-packages repository. If that isn't feasible, then hourly or semi-hourly test runs will help catch problems sooner.
+
+Aside: The quality of curated packages relies upon the quality of our end-to-end tests. Correspondingly, the more often we run them, the faster we learn of, and can fix, problems. If we're not able to run the full battery of tests for each PR, then it is strongly recommended that we do whatever is in our power to be able to do so.
+
+#### Test Configuration Control
+
+A means by which tests can be disabled or enabled based upon their properties, requirements, or current statuses. In plainer words, a way to write a test that doesn't run for a specific provider, or to temporarily disable tests for specific host OS. This input, along with the test manifest are envisioned as being the bulk of the input to a given CP E2E test run.
+
+#### End-To-End Test Documentation
+
+Documentation will be generated that, in addition to the forthcoming design document, will aid developers and other contributors understand the end-to-end test framework, specifically how to write a new end-to-end test, how to run an end-to-end test outside of the build pipelines, how to be notified of failures, and how to debug test failures.
+
+A stretch goal could be to add documentation-style tests in the spirit of [Rust's documentation tests](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html) or [go test examples](https://pkg.go.dev/testing#hdr-Examples).
+
```

Comment:
One other thing I can think of is to enhance test logging and notification logics so that the troubleshooting process for developers become easier.

Reply:
Dan spent a fair amount of time creating tools to extract test information. Hopefully, we don't go with a system where we need to develop similar tools

Reply:
> Dan spent a fair amount of time creating tools to extract test information. Hopefully, we don't go with a system where we need to develop similar tools

I'm not sure of all the tools you might be referencing, but at least some of the ones I'm aware of are a result of cloudwatch being a little awkward to use... I'm not sure how we avoid that, but your advice is good nonetheless.